src/traffic_ctl/host.cc

```diff
@@ -53,40 +66,75 @@ static int
 status_down(unsigned argc, const char **argv)
 {
   int down_time     = 0;
+  char *reason      = nullptr;
   const char *usage = "host down HOST [OPTIONS]";
 
   const ArgumentDescription opts[] = {
-    {"time", 'I', "number of seconds that a host is marked down", "I", &down_time, nullptr, nullptr}};
+    {"time", 'I', "number of seconds that a host is marked down", "I", &down_time, nullptr, nullptr},
+    // memory is allocated for 'reason', if this option is used
+    {"reason", '-', "reason for marking the host down, one of 'manual|active|local'", "S*", &reason, nullptr, nullptr},
+  };
 
   if (!CtrlProcessArguments(argc, argv, opts, countof(opts)) || n_file_arguments < 1) {
     return CtrlCommandUsage(usage, opts, countof(opts));
   }
 
+  // if reason is not set, set it to manual (default)
+  if (reason == nullptr) {
+    reason = ats_strdup(MANUAL.c_str());
```

Comment:
Why call `ats_strdup`?

Reply:
I only call str_dup if the --reason option is not used and then default it to manual.  In all other cases when the option --reason is used, CtrlProcessAruguments() allocates (str_dup) memory for the 'reason' pointer setting it to the command line argument.  So, I have to call ats_free regardless onece 'reason' is used.